hera_pspec/grouping.py

```diff
@@ -386,6 +412,15 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
                     wgts = uvp.get_wgts(spw, blp, p)
                     ints = uvp.get_integrations(spw, blp, p)[:, None]
                     w = (ints * np.sqrt(nsmp))
+
+                    # Get error bar weights and set invalid ones to zero.
+                    errws = {}
+                    for stat in stat_l:
+                        errs = uvp.get_stats(stat, (spw, blp, p))
```

Comment:
This fix is now more efficient, but it still doesn't address the case when `errs` has negative, but `!=-99` data. This is why I think its better to use `no_data = errs < 0`, which will also capture when `errs == -99`

Reply:
We should set invalid values to `np.nan`, shouldn't we?

Reply:
We could use `nan`s. Perhaps that is a safer solution. In that case, @duncanjmr, your indexing arrays would become `no_data = np.isnan(errs)`.

Reply:
Also, @duncanjmr, because you are indexing a mutable variable, you will want to ensure you aren't accidently overwriting the parent `uvp` object by using
```
errs = uvp.get_stats(stat, (spw, blp, p)).copy()
```